### ISSUE TITLE
Remove husky dependency?

### ISSUE THREAD

**Issue by rjuju** [powa-team/powa-web#244]
@pgiraud 

I see that husky is creating a git pre-commit script and overriding developpers git configuration to call it automatically by stting core.hooksPath.

I'm not a fan of this approach for multiple reasons:
- in general dependencies should have no business changing your git configuration behding your back
- it also makes your life complicated if you want to configure your own hooks (for instance if you want to run ruff or anything else)
- apparently the various npm checks are only checked in this pre-commit hook but not in the CI, which seems like a bad idea.

I think that we could entirely remove the husky dependency, which doesn't really solve any problem anymore, and instead provide some pre-commit script with some documentation so that users are free to configure it, and call that script in the CI.

**Comment by pgiraud** [comment]
I agree, I can do that.

**Comment by rjuju** [comment]
thanks!

**Comment by pgiraud** [comment]
> * apparently the various npm checks are only checked in this pre-commit hook but not in the CI, which seems like a bad idea

This is wrong. npm checks are done in CI via https://github.com/powa-team/powa-web/blob/master/.github/workflows/javascript.yml
